### PR TITLE
Set up the correct remoting type for id/fk

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @ritch 

Connect to #1221

### CODE REVIEW

File: test/util/model-tests.js [hunk 1]
```diff
@@ -169,11 +170,11 @@ module.exports = function defineModelTestsWithDataSource(options) {
 
     describe('Model.upsert(data, callback)', function() {
       it('Update when record with id=data.id found, insert otherwise', function(done) {
-        User.upsert({first: 'joe', id: 7}, function(err, user) {
+        User.upsert({first: 'joe', id: '7'}, function(err, user) {
```
Comment (ritch) [inline test/util/model-tests.js]: ...? why does this have to change

**Comment by ritch** [comment]
@raymondfeng 1 minor change... not comfortable merging this when you are changing the tests.

**Comment by raymondfeng** [comment]
I found some issues related to how loopback-connector-remote sets up the models. As the PR defers the remoting setup after a model is attached to the data source, the loopback-connector-remote fails as it has some assumptions about the timing. Let's chat more.

**Comment by ritch** [comment]
@raymondfeng we can chat offline if you want. Just ping me today.

File: lib/model.js [hunk 2]
```diff
@@ -98,6 +98,34 @@ var stringUtils = require('underscore.string');
 
 var Model = module.exports = registry.modelBuilder.define('Model');
 
+function getRemotingType(prop) {
+  if (prop.type === Number) {
+    return 'number';
+  }
+  if (prop.type === Boolean) {
+    return 'number';
```
Comment (ritch) [inline lib/model.js]: This should be boolean @raymondfeng

**Comment by bajtos** [comment]
@raymondfeng @ritch what's the status of this patch?

**Comment by ritch** [comment]
Not sure... @raymondfeng ?

**Comment by fabien** [comment]
This is a critical issue for many users - why hasn't this been merged yet? I read about failing tests in `loopback-connector-remote`, but why has this been preventing anyone to prioritize and fix this?

/cc @raymondfeng

**Comment by raymondfeng** [comment]
The patch fixes the issue but it breaks the assumption of loopback-connector-remote which expects the remoting metadata to be available earlier. We need to rebase the fix and have @ritch look into the loopback-connector-remote issue.

**Comment by fabien** [comment]
@raymondfeng @ritch @bajtos - any updates?

File: lib/model.js (hunk 2 above)
Comment (bajtos) [inline lib/model.js]: @raymondfeng please add a unit-test for this case, to make sure we don't introduce a similar bug in the future.

File: lib/model.js [hunk 3]
```diff
@@ -164,10 +190,41 @@ Model.setup = function() {
     }
   };
 
+  // resolve relation functions
+  sharedClass.resolve(function resolver(define) {
+
```
Comment (superkhau) [inline lib/model.js]: extra space here

File: test/util/model-tests.js [hunk 4]
```diff
@@ -169,11 +170,11 @@ module.exports = function defineModelTestsWithDataSource(options) {
...
           assert(!err);
           assert.equal(user.first, 'joe');
 
-          User.upsert({first: 'bob', id: 7}, function(err, updatedUser) {
+          User.upsert({first: 'bob', id: '7'}, function(err, updatedUser) {
```
Comment (superkhau) [inline test/util/model-tests.js]: same here why string now

**Comment by bajtos** [comment]
I am closing this pull request as abandoned. If there is anybody willing to take over this work in progress and make the changes requested in the comments above, then please open a new pull request.

**Pull request closed without merging**